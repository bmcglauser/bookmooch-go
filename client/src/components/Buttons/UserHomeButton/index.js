import UserHomeButton from './UserHomeButton';

export default UserHomeButton;