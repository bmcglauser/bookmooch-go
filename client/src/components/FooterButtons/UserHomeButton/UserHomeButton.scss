@use '../../../App' as *;

.user-button-wrapper {
  margin: 15px 5px 0px 5px;
  width: 75px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  
  .user-button {
    height: 35px;
    width: 35px;
    border: none;
    border-radius: 5px;
    background-image: url(../../../assets/user-icon.png);
    background-repeat: round;

    &:focus {
      outline: none;
    }
  }

  p {
    margin-top: 5px;
    color: $off-white;
    font-size: 10pt;
  }

}