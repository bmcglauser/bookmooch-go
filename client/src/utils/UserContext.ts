import { createContext } from 'react';

const UserContext = createContext<any>({username: '', setUsernameCB: ()=>{return }});

export default UserContext;