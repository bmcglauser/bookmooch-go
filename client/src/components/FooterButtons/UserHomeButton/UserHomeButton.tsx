import React, { FunctionComponent, useContext } from 'react';
import './UserHomeButton.scss';
import { Link } from 'react-router-dom';
import UserContext from '../../../utils/UserContext';

const UserHomeButton: FunctionComponent = () => {
  const { username } = useContext(UserContext);

  return (
    <Link to={`/profile/${username}`} style={{textDecoration:'none'}}>
      <div className="user-button-wrapper">
        <button className="user-button" />
        <p>User Home</p>
      </div>
    </Link>
  );}

export default UserHomeButton;