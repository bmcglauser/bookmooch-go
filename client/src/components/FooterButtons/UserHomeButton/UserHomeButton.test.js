import React from 'react';
import { render } from '@testing-library/react';

import UserHomeButton from './';
import { MemoryRouter } from 'react-router-dom';
import UserContext from '../../../utils/UserContext';

describe('UserHomeButton', () => {
  it('should render with UserContext', () => {
    const tree = render(
      <UserContext.Provider value={{ username: 'mattyboi' }}>
        <UserHomeButton />
      </UserContext.Provider>,
      { wrapper: MemoryRouter }
    );
    expect(tree).toMatchSnapshot();
  });
});
