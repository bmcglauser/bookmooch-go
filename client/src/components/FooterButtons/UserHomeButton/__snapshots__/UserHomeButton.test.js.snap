// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHomeButton should render with UserContext 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/profile/mattyboi"
        style="text-decoration: none;"
      >
        <div
          class="user-button-wrapper"
        >
          <button
            class="user-button"
          />
          <p>
            User Home
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/profile/mattyboi"
      style="text-decoration: none;"
    >
      <div
        class="user-button-wrapper"
      >
        <button
          class="user-button"
        />
        <p>
          User Home
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
